### ISSUE TITLE
Can't remove Kobo DRM with DeDRM or Obok

### ISSUE THREAD

**Issue by jDonut562** [noDRM/DeDRM_tools#376]
### Question / bug report

I've been unable to decrypt a kobo book that I bought. It's a Japanese manga. I've tried with both the Obok plugin, and the regular DeDRM plugin (using ADE.) Both methods seem to fail to remove the DRM. I'm not sure if this book just has some new style of encryption that hasn't been broken yet, or if I'm just doing something wrong. I'm using Windows 10. I'm able to open the book in the kobo app and in ADE.

When I first installed DeDRM, I manually added a key entry for the ADE settings. I saw in other issues that this can cause problems, so I've already tried removing the key entry, restarting calibre, and letting DeDRM do it automatically. It doesn't seem to remove the DRM either way.

I've also tried the Obok plugin for the main Kobo version of the ebook, and while it does automatically detect that I have the ebook on my computer, it still doesn't seem to be able to decrypt the book. Unlike the ADE version, the title of the book, once added to calibre, is just random characters. It's a different set of random characters every time I try to decrypt the book.

I also tried installing an older version of calibre and the plugins. I used calibre 4.23 and DeDRM v6.8.0, and it did pretty much exactly the same thing.

The logs I've posted are for calibre 6.23 and DeDRM v10.0.3. Any help would be very much appreciated.

### Which version of Calibre are you running?

6.23

### Which version of the DeDRM plugin are you running?

v10.0.3

### Log output

Log from DeDRM (ADE) attempt:
```log
calibre, version 6.23.0
ERROR: Loading book failed: Failed to open the book at C:\Users\Meta\Calibre Library\Unknown\ruridoragon 1 (11)\ruridoragon 1 - Unknown.epub. Click "Show details" for more info.

Failed to convert book: C:\Users\Meta\Calibre Library\Unknown\ruridoragon 1 (11)\ruridoragon 1 - Unknown.epub with error:
Traceback (most recent call last):
  File "calibre\customize\ui.py", line 469, in get_file_type_metadata
  File "calibre\customize\builtins.py", line 175, in get_metadata
  File "calibre\ebooks\metadata\epub.py", line 237, in get_quick_metadata
  File "calibre\ebooks\metadata\epub.py", line 217, in get_metadata
  File "calibre\ebooks\metadata\epub.py", line 155, in read_bytes
  File "calibre\utils\zipfile.py", line 1005, in read
  File "calibre\utils\zipfile.py", line 1033, in open
  File "calibre\utils\zipfile.py", line 994, in getinfo
KeyError: "There is no item named 'OPS/advanced.opf' in the archive"
DeDRM v10.0.3: Trying to decrypt ruridoragon 1 - Unknown.epub
DeDRM v10.0.3: Verifying zip archive integrity
DeDRM v10.0.3: Post-processing took 0.0 seconds
DeDRM v10.0.3: Finished after 1.0 seconds
InputFormatPlugin: EPUB Input running
on C:\Users\Meta\AppData\Local\Temp\calibre_9pm2qjty\6jyey6uh.epub
Failed to run pipe worker with command: from calibre.srv.render_book import viewer_main; viewer_main()
Traceback (most recent call last):
  File "runpy.py", line 196, in _run_module_as_main
  File "runpy.py", line 86, in _run_code
  File "site.py", line 83, in <module>
  File "site.py", line 78, in main
  File "site.py", line 50, in run_entry_point
  File "calibre\utils\ipc\worker.py", line 196, in main
  File "<string>", line 1, in <module>
  File "calibre\srv\render_book.py", line 826, in viewer_main
  File "calibre\srv\render_book.py", line 817, in render_for_viewer
  File "calibre\srv\render_book.py", line 795, in render
  File "calibre\srv\render_book.py", line 608, in process_exploded_book
  File "calibre\srv\render_book.py", line 263, in create_cover_page
  File "calibre\srv\render_book.py", line 216, in find_epub_cover
  File "calibre\ebooks\oeb\polish\cover.py", line 284, in find_cover_image_in_page
  File "src/lxml/xpath.pxi", line 430, in lxml.etree.XPath.__call__
  File "src/lxml/apihelpers.pxi", line 41, in lxml.etree._documentOrRaise
TypeError: Invalid input object: bytes
```

Log from Obok attempt:
```log
calibre, version 6.23.0
ERROR: Loading book failed: Failed to open the book at C:\Users\Meta\Calibre Library\Unknown\ht1lm6x1 (12)\ht1lm6x1 - Unknown.epub. Click "Show details" for more info.

Failed to convert book: C:\Users\Meta\Calibre Library\Unknown\ht1lm6x1 (12)\ht1lm6x1 - Unknown.epub with error:
Traceback (most recent call last):
  File "calibre\customize\ui.py", line 469, in get_file_type_metadata
  File "calibre\customize\builtins.py", line 175, in get_metadata
  File "calibre\ebooks\metadata\epub.py", line 237, in get_quick_metadata
  File "calibre\ebooks\metadata\epub.py", line 217, in get_metadata
  File "calibre\ebooks\metadata\epub.py", line 155, in read_bytes
  File "calibre\utils\zipfile.py", line 1005, in read
  File "calibre\utils\zipfile.py", line 1033, in open
  File "calibre\utils\zipfile.py", line 994, in getinfo
KeyError: "There is no item named 'OPS/advanced.opf' in the archive"
DeDRM v10.0.3: Trying to decrypt ht1lm6x1 - Unknown.epub
DeDRM v10.0.3: Verifying zip archive integrity
DeDRM v10.0.3: Post-processing took 0.0 seconds
DeDRM v10.0.3: Finished after 1.0 seconds
InputFormatPlugin: EPUB Input running
on C:\Users\Meta\AppData\Local\Temp\calibre_g1s30jct\oorx_2ct.epub
Failed to run pipe worker with command: from calibre.srv.render_book import viewer_main; viewer_main()
Traceback (most recent call last):
  File "runpy.py", line 196, in _run_module_as_main
  File "runpy.py", line 86, in _run_code
  File "site.py", line 83, in <module>
  File "site.py", line 78, in main
  File "site.py", line 50, in run_entry_point
  File "calibre\utils\ipc\worker.py", line 196, in main
  File "<string>", line 1, in <module>
  File "calibre\srv\render_book.py", line 826, in viewer_main
  File "calibre\srv\render_book.py", line 817, in render_for_viewer
  File "calibre\srv\render_book.py", line 795, in render
  File "calibre\srv\render_book.py", line 608, in process_exploded_book
  File "calibre\srv\render_book.py", line 263, in create_cover_page
  File "calibre\srv\render_book.py", line 216, in find_epub_cover
  File "calibre\ebooks\oeb\polish\cover.py", line 284, in find_cover_image_in_page
  File "src/lxml/xpath.pxi", line 430, in lxml.etree.XPath.__call__
  File "src/lxml/apihelpers.pxi", line 41, in lxml.etree._documentOrRaise
TypeError: Invalid input object: bytes
```

**Comment by ElleKayEm** [comment]
Both logs just show DeDRM, not Obok.  At any rate, seems to be an issue with the book itself:

> KeyError: "There is no item named 'OPS/advanced.opf' in the archive"

Unzip it and see if the contents are readable.  If it's just full-page scans and the images are decrypted, then you could extract them and make a cbz file (which could be converted).

**Comment by jDonut562** [comment]
Ah, you're right. The images inside the epub got decrypted so I can read/extract them. Thanks for your help!

Sorry for the confusing logs. I forgot to mention that I didn't get errors upon importing/decrypting the book, it's just when I tried to open and read the decrypted book in calibre that it would give the errors I posted. I guess it doesn't matter now, but I thought I'd mention it just in case.

Anyway, thanks again!
